CRM/CloseAccountingPeriod/Form/PriorFinancialPeriod.php

```diff
@@ -61,9 +68,9 @@ public function setDefaultValues() {
   public function buildQuickForm() {
     $priorFinancialPeriod = CRM_CloseAccountingPeriod_BAO_CloseAccountingPeriod::getPriorFinancialPeriod($this->_contactID);
     if ($priorFinancialPeriod) {
-      $this->assign('priorFinancialPeriod', $priorFinancialPeriod);
+      $this->assign('priorFinancialPeriod', $priorFinancialPeriod['M'] . '-' . $priorFinancialPeriod['Y']);
     }
-    $this->addDate('prior_financial_period', ts('Prior Financial Period'), TRUE);
+    $this->add('date', 'prior_financial_period', ts('Prior Financial Period'), CRM_Core_SelectValues::date(NULL, 'M Y', 2, 5), TRUE);
```

Comment:
Again, please use formatting from civicrm/admin/setting/date?reset=1 Month and Year field.